lib/preprocessor.js

```diff
@@ -36,6 +37,94 @@ var createNextProcessor = function (preprocessors, file, done) {
   }
 }
 
+/* Merge two lists, removing duplicates, and doing everything possible to
+ * maintain the order of the two lists.
+ *
+ * This function guarantees that the order of list1 is preserved (that is, if x
+ * comes before y in list1, x comes before y in the returned list) and tries not
+ * to undo the order of list2, though sometimes it is unavoidable.
+ *
+ * For example, if we have list1 = [1, 2, 4] and list2 = [2, 1, 3, 4], then the
+ * merged list would be [1, 2, 3, 4], since that preserves the order of list1
+ * while doing the best job possible of preserving the order of list2.
+ *
+ * A case like list1 = [1, 3], list2 = [3, 2, 1] is more complicated.  It's not
+ * clear what the best merged list is, but it's probably either [2, 1, 3] or
+ * [1, 3, 2].
+ *
+ * In general, it's not totally clear what the "best" merged list is, but there
+ * are some basic properties that anyone would expect:
+ *   - Since the order of list1 is preserved, the merged list will look like
+ *     list1 with the elements exclusive to list2 inserted in betweeen
+ *   - If list2[i] is not in list1, and it is possible to insert list2[i] into
+ *     list1 without contradicting the order of list2, then it should be
+ *     inserted in such a way
+ */
+var mergeLists = function (list1, list2) {
+  /* This is going to get mathematical.  As noted above, the merged list will be
+   * a copy of list1 with the items exclusive to list2 inserted in between.  But
+   * additionally, we want to preserve the order of list2 as much as possible.
+   * In more formal terms, for all x and y from list2, we want to minimize the
+   * number of times that x is before y in the merged list but after y in list2.
+   * We call each such time an "inversion", after the term in discrete math.
+   *
+   * We are going to take a greedy approach to this:
+   *
+   * merged_list = a copy of list1
+   * for(i = 0; i < list2.length; i++)
+   *   if(list2[i] is not in list1)
+   *     insert list2[i] into merged_list in the earliest place that creates the
+   *     minimum number of inversions
+   * return merged_list
+   *
+   * It can be proven that this gives you the two properties mentioned in the
+   * header comment above
+   *
+   * For those who care, this is very slow, and taking
+   * O(list2.length*list2.length*(list1.length + list2.length)) time.
+   */
+  var merged = list1.slice() // The merged list to return
+  var mergedIndexes = _.invert(merged)
+  for (var i = 0; i < list2.length; i++) {
+    var elem = list2[i]
+    if (mergedIndexes[elem] === undefined) {
+      // Count the inversions for every possible insertion position
+      var inversionCnts, j, k
+      if (typeof Int32Array !== 'undefined') {
```

Comment:
It's a lot faster.  It forces all the following arithmetic to be integer arithmetic instead of floating point.    It's the inner loop of a potential bottleneck, so I thought it might be a good idea.  That said, unless someone had like 15 processors running on the same file, it probably wouldn't be noticeable.